doc/sphinx/user-extensions/syntax-extensions.rst

```diff
@@ -7,64 +7,179 @@ In this chapter, we introduce advanced commands to modify the way Coq
+   .. coqtop:: none
 
-   Specifying :token:`scope_name` associates the notation with that scope.  Otherwise
-   it is a :gdef:`lonely notation`, that is, not associated with a scope.
+      Require Import List.
+      Require Import Relations.
+      Set Printing Notations.
 
-   .. todo indentation of this chapter is not consistent with other chapters.  Do we have a standard?
+   .. coqtop:: in
 
-For example, the following definition permits using the infix expression :g:`A /\ B`
-to represent :g:`(and A B)`:
+      Notation Nlist := (list nat).
 
-.. coqtop:: in
+   .. coqtop:: all
 
-   Notation "A /\ B" := (and A B).
+      Check 1 :: 2 :: 3 :: nil.
 
-:g:`"A /\ B"` is a *notation*, which tells how to represent the abbreviated term
-:g:`(and A B)`.
+   .. coqtop:: in
+
+      Notation reflexive R := (forall x, R x x).
+
+   .. coqtop:: all
+
+      Check forall A:Prop, A <-> A.
+      Check reflexive iff.
+
+   .. coqtop:: in
+
+      Notation Plus1 B := (Nat.add B 1).
+
+   .. coqtop:: all
+
+      Compute (Plus1 3).
+
+   An abbreviation expects no precedence nor associativity, since it
+   is parsed as a usual application. Abbreviations are used as
+   much as possible by the Coq printers unless the modifier ``(only
+   parsing)`` is given.
+
+   An abbreviation is bound to an absolute name as an ordinary definition is
+   and it also can be referred to by a qualified name.
+
+   Abbreviations are syntactic in the sense that they are bound to
+   expressions which are not typed at the time of the definition of the
+   abbreviation but at the time they are used. Especially, abbreviations
+   can be bound to terms with holes (i.e. with “``_``”). For example:
+
+   .. coqtop:: none reset
+
+      Set Strict Implicit.
+      Set Printing Depth 50.
+
+   .. coqtop:: in
+
+      Definition explicit_id (A:Set) (a:A) := a.
+
+   .. coqtop:: in
+
+      Notation id := (explicit_id _).
+
+   .. coqtop:: all
+
+      Check (id 0).
+
+   Abbreviations disappear when a section is closed. No typing of the
+   denoted expression is performed at definition time. Type checking is
+   done only at the time of use of the abbreviation.
+
+   Like for notations, if the right-hand side of an abbreviation is a
+   partially applied constant, the abbreviation inherits the implicit
+   arguments and notation scopes of the constant. As an
+   exception, if the right-hand side is just of the form :n:`@@qualid`,
+   this conventionally stops the inheritance of implicit arguments.
+
+   Like for notations, it is possible to bind binders in
+   abbreviations. Here is an example:
+
+   .. coqtop:: in reset
+
+      Definition force2 q (P:nat*nat -> Prop) :=
+        (forall n', n' >= fst q -> forall p', p' >= snd q -> P q).
+
+      Notation F p P := (force2 p (fun p => P)).
+      Check exists x y, F (x,y) (x >= 1 /\ y >= 2).
+
+.. _ReservingNotations:
+
+Reserving notations
+-------------------
+
+.. cmd:: Reserved Notation @string {? ( {+, @syntax_modifier } ) }
```

Comment:
Your comment is valid. I get @proux01's point though: by doing so we separate the "lexing+parsing" phase from the interpretation. Maybe we can keep it that way and add simple examples in the introduction.

Reply:
Not that there is actually no such thing as `defining notations without reserving them`, it's just that the `Notation` command currently silently reserves notations for you if it wasn't already done.

Indeed, there are very good rationales for making the parsing / interpretation phase separation more explicit (case in point, your comments in this PR indicate that you didn't fully got it with the previous state of the manual). Not just the fact that the phases are in practice separated but also the fact that they are of very different nature in terms of compositionallity and risks. The reserving notations / parsing things is one of the least modular aspects of Coq. An unfortunate grammar extension can easily wreak havoc in code importing it and seriously compromise compositionality between diferent libraries (although the import categories enabling to import a module without its grammar extensions is a very nice workaround to make progress in that direction). In contrary notation interpretation is much easier to override thanks to the scope mechanism and doesn't present nearly as much risk in terms of compositionality.

I agree however that maybe we should add more examples, potentially using the `Notation` command, in the reserving notation section, so as to be able to fully demonstrate parsing and printing results.

Reply:
> Not that there is actually no such thing as `defining notations without reserving them`, it's just that the `Notation` command currently silently reserves notations for you if it wasn't already done.

No.  Think in terms of what the user has to do, not in terms of internal operations.  They can define notations with a single command.  If they want to overload the syntax, they must do a reserve first.

> An unfortunate grammar extension can easily wreak havoc in code importing it

This applies equally to notations defined solely with Notation and those defined with Reserved Notation.

> case in point, your comments in this PR indicate that you didn't fully got it with the previous state of the manual

This comment is not helpful.  If you think I'm mistaken, you should discuss it with me and explain the rationale rather then writing me off.  I have no idea what you think I'm mistaken about.  I hope our comments here are considered a discussion aimed at reaching a consensus (and sometimes mutual education) rather than an argument to win or lose.